diff --git conformance/BUILD conformance/BUILD
index 6f31a01dc..07a940a7b 100644
--- conformance/BUILD
+++ conformance/BUILD
@@ -12,21 +12,18 @@ load("@rules_cc//cc:cc_test.bzl", "cc_test")
 load("@rules_cc//cc:defs.bzl", "cc_binary", "objc_library")
 load("@rules_java//java:defs.bzl", "java_binary")
 load("@rules_python//python:py_binary.bzl", "py_binary")
-load("@rules_ruby//ruby:defs.bzl", "rb_binary")
 load(
     "//:protobuf.bzl",
     "internal_csharp_proto_library",
     "internal_objc_proto_library",
     "internal_php_proto_library",
     "internal_py_proto_library",
-    "internal_ruby_proto_library",
 )
 load("//bazel:cc_proto_library.bzl", "cc_proto_library")
 load("//bazel:java_lite_proto_library.bzl", "java_lite_proto_library")
 load("//bazel:java_proto_library.bzl", "java_proto_library")
 load("//bazel:proto_library.bzl", "proto_library")
 load("//build_defs:internal_shell.bzl", "inline_sh_binary")
-load("//ruby:defs.bzl", "internal_ruby_proto_library")
 
 exports_files([
     "bazel_conformance_test_runner.sh",
@@ -420,15 +417,6 @@ internal_php_proto_library(
     ],
 )
 
-internal_ruby_proto_library(
-    name = "conformance_ruby_proto",
-    srcs = ["conformance.proto"],
-    visibility = [
-        "//conformance:__pkg__",
-        "//ruby:__pkg__",
-    ],
-)
-
 cc_binary(
     name = "conformance_cpp",
     testonly = 1,
@@ -619,22 +607,6 @@ cc_binary(
     deps = [":conformance_objc_lib"],
 )
 
-rb_binary(
-    name = "conformance_ruby",
-    testonly = True,
-    srcs = ["conformance_ruby.rb"],
-    main = "conformance_ruby.rb",
-    visibility = ["//ruby:__subpackages__"],
-    deps = [
-        ":conformance_ruby_proto",
-        "//conformance/test_protos:test_messages_edition2023_ruby_proto",
-        "//conformance/test_protos:test_messages_edition_unstable_ruby_proto",
-        "//ruby:conformance_editions_test_ruby_proto",
-        "//ruby:conformance_test_ruby_proto",
-        "//ruby:protobuf",
-    ],
-)
-
 ################################################################################
 # Distribution files
 ################################################################################
diff --git conformance/test_protos/BUILD conformance/test_protos/BUILD
index a4788b8a6..2d3b23c4f 100644
--- conformance/test_protos/BUILD
+++ conformance/test_protos/BUILD
@@ -8,7 +8,6 @@ load("//bazel:cc_proto_library.bzl", "cc_proto_library")
 load("//bazel:java_lite_proto_library.bzl", "java_lite_proto_library")
 load("//bazel:java_proto_library.bzl", "java_proto_library")
 load("//bazel:proto_library.bzl", "proto_library")
-load("//ruby:defs.bzl", "internal_ruby_proto_library")
 
 package(
     default_testonly = True,
@@ -118,13 +117,3 @@ internal_objc_proto_library(
     name = "test_messages_edition_unstable_objc_proto",
     srcs = [":test_messages_edition_unstable.proto"],
 )
-
-internal_ruby_proto_library(
-   name = "test_messages_edition2023_ruby_proto",
-   srcs = [":test_messages_edition2023.proto"],
-)
-
-internal_ruby_proto_library(
-    name = "test_messages_edition_unstable_ruby_proto",
-    srcs = [":test_messages_edition_unstable.proto"],
-)
\ No newline at end of file
