diff --git a/fetch.bzl b/fetch.bzl
index 5baeeb1..cf148bd 100644
--- fetch.bzl
+++ fetch.bzl
@@ -195,6 +195,14 @@ def fetch_images():
         image = "nvcr.io/nvidia/k8s-device-plugin",
     )
 
+    # Helm chart lacking `mediaType` in manifest.
+    # https://github.com/bazel-contrib/rules_oci/issues/839
+    oci_pull(
+        name = "envoyproxy_gateway_helm",
+        digest = "sha256:905eced000d4b2acb78f802f5d03af32a08d30478808c20d522ffa735476bc5d",
+        image = "docker.io/envoyproxy/gateway-helm",
+    )
+
     _DEB_TO_LAYER = """\
 genrule(
     name = "layer",
diff --git a/oci/private/pull.bzl b/oci/private/pull.bzl
index 5f1e9c5..748a473 100644
--- oci/private/pull.bzl
+++ oci/private/pull.bzl
@@ -55,17 +55,20 @@ Unable to retrieve the image manifest. This could be due to
 *) If there is a configured URL Rewriter, check that it does not block the request.
 """
 
+_DEFAULT_MANIFEST_MEDIA_TYPE = "application/vnd.oci.image.manifest.v1+json"
+_DEFAULT_INDEX_MEDIA_TYPE = "application/vnd.oci.image.index.v1+json"
+
 # Supported media types
 # * OCI spec: https://github.com/opencontainers/image-spec/blob/main/media-types.md
 # * Docker spec: https://github.com/distribution/distribution/blob/main/docs/spec/manifest-v2-2.md#media-types
 _SUPPORTED_MEDIA_TYPES = {
     "index": [
         "application/vnd.docker.distribution.manifest.list.v2+json",
-        "application/vnd.oci.image.index.v1+json",
+        _DEFAULT_INDEX_MEDIA_TYPE,
     ],
     "manifest": [
         "application/vnd.docker.distribution.manifest.v2+json",
-        "application/vnd.oci.image.manifest.v1+json",
+        _DEFAULT_MANIFEST_MEDIA_TYPE,
     ],
 }
 
@@ -74,6 +77,23 @@ _DOWNLOAD_HEADERS = {
     "Docker-Distribution-API-Version": "registry/2.0",
 }
 
+def _determine_media_type(manifest):
+    """Determine the media type of a manifest.
+
+    In the absence of an explicit media type, infer a default value based on other properties.
+    Image manifests have a required `config` field:
+    https://specs.opencontainers.org/image-spec/manifest/#image-manifest-property-descriptions.
+    Image indices have a required `manifests` field:
+    https://specs.opencontainers.org/image-spec/image-index/#image-index-property-descriptions
+    """
+    if "mediaType" in manifest:
+        return manifest["mediaType"]
+    if "config" in manifest:
+        return _DEFAULT_MANIFEST_MEDIA_TYPE
+    if "manifests" in manifest:
+        return _DEFAULT_INDEX_MEDIA_TYPE
+    fail("Unable to determine media type")
+
 def _config_path(rctx):
     if rctx.attr.config:
         return rctx.path(rctx.attr.config)
@@ -204,12 +224,13 @@ def _oci_pull_impl(rctx):
     downloader = _create_downloader(rctx, au)
 
     manifest, size, digest = downloader.download_manifest(rctx.attr.identifier, "manifest.json")
+    media_type = _determine_media_type(manifest)
 
-    if manifest["mediaType"] in _SUPPORTED_MEDIA_TYPES["manifest"]:
+    if media_type in _SUPPORTED_MEDIA_TYPES["manifest"]:
         # copy manifest.json to blobs with its digest.
         rctx.template(_digest_into_blob_path(digest), "manifest.json")
 
-    elif manifest["mediaType"] in _SUPPORTED_MEDIA_TYPES["index"]:
+    elif media_type in _SUPPORTED_MEDIA_TYPES["index"]:
         # image index manifest: download the image manifest for the target platform.
         if not rctx.attr.platform:
             fail("{}/{} is a multi-architecture image, so attribute 'platforms' is required.".format(
@@ -248,11 +269,12 @@ def _oci_pull_impl(rctx):
         # ```
         # See https://github.com/bazel-contrib/rules_oci/pull/596 for more details on this race codition.
         manifest, size, digest = downloader.download_manifest(matching_manifest["digest"], "platform-manifest.json")
+        media_type = _determine_media_type(manifest)
 
         # copy platform-manifest.json to blobs with its digest.
         rctx.template(_digest_into_blob_path(digest), "platform-manifest.json")
     else:
-        fail("Unrecognized mediaType {} in manifest file".format(manifest["mediaType"]))
+        fail("Unrecognized mediaType {} in manifest file".format(media_type))
 
     config_output_path = _digest_into_blob_path(manifest["config"]["digest"])
     downloader.download_blob(manifest["config"]["digest"], config_output_path, block = True)
@@ -282,7 +304,7 @@ def _oci_pull_impl(rctx):
             r.wait()
 
     rctx.file("index.json", util.build_manifest_json(
-        media_type = manifest["mediaType"],
+        media_type = media_type,
         size = size,
         digest = digest,
         platform = rctx.attr.platform,
@@ -359,17 +381,18 @@ def _oci_alias_impl(rctx):
     available_platforms = []
 
     manifest, _, digest = downloader.download_manifest(rctx.attr.identifier, "mf.json")
+    media_type = _determine_media_type(manifest)
     rctx.file("digest.txt", digest)
 
     if rctx.attr.platforms:
-        if manifest["mediaType"] in _SUPPORTED_MEDIA_TYPES["index"]:
+        if media_type in _SUPPORTED_MEDIA_TYPES["index"]:
             # multi arch image.
             for submanifest in manifest["manifests"]:
                 parts = [submanifest["platform"]["os"], submanifest["platform"]["architecture"]]
                 if "variant" in submanifest["platform"]:
                     parts.append(submanifest["platform"]["variant"])
                 available_platforms.append('"{}"'.format("/".join(parts)))
-        elif manifest["mediaType"] in _SUPPORTED_MEDIA_TYPES["manifest"]:
+        elif media_type in _SUPPORTED_MEDIA_TYPES["manifest"]:
             # single arch image where the user specified the platform.
             config_output_path = _digest_into_blob_path(manifest["config"]["digest"])
             downloader.download_blob(manifest["config"]["digest"], config_output_path, block = True)
diff --git a/oci/tests/BUILD.bazel b/oci/tests/BUILD.bazel
index 17abd57..fb4996f 100644
--- oci/tests/BUILD.bazel
+++ oci/tests/BUILD.bazel
@@ -79,6 +79,7 @@ build_test(
         "@es_kibana_image",
         "@quay_clair_image",
         "@nvidia_k8s_device_plugin_image",
+        "@envoyproxy_gateway_helm",
     ],
 )
 
